espnet/nets/pytorch_backend/wavenet.py

```diff
@@ -0,0 +1,381 @@
+        if n_samples > h.shape[2]:
+            h = F.pad(h, (0, n_samples - h.shape[2]), "replicate")
+
+        # padding if the length less than
+        n_pad = self.receptive_field - x.size(1)
+        if n_pad > 0:
+            x = F.pad(x, (n_pad, 0), "constant", self.n_quantize // 2)
+            h = F.pad(h, (n_pad, 0), "replicate")
+
+        # prepare buffer
+        output = self._preprocess(x)
+        h_ = h[:, :, :x.size(1)]
+        output_buffer = []
+        buffer_size = []
+        for l, d in enumerate(self.dilations):
+            output, _ = self._residual_forward(
+                output, h_, self.dil_sigmoid[l], self.dil_tanh[l],
+                self.aux_1x1_sigmoid[l], self.aux_1x1_tanh[l],
+                self.skip_1x1[l], self.res_1x1[l])
+            if d == 2 ** (self.dilation_depth - 1):
+                buffer_size.append(self.kernel_size - 1)
+            else:
+                buffer_size.append(d * 2 * (self.kernel_size - 1))
+            output_buffer.append(output[:, :, -buffer_size[l] - 1: -1])
+
+        # generate
+        samples = x[0]
+        start = time.time()
+        for i in range(n_samples):
+            output = samples[-self.kernel_size * 2 + 1:].unsqueeze(0)
+            output = self._preprocess(output)
+            h_ = h[:, :, samples.size(0) - 1].contiguous().view(1, self.n_aux, 1)
+            output_buffer_next = []
+            skip_connections = []
+            for l, d in enumerate(self.dilations):
+                output, skip = self._generate_residual_forward(
+                    output, h_, self.dil_sigmoid[l], self.dil_tanh[l],
+                    self.aux_1x1_sigmoid[l], self.aux_1x1_tanh[l],
+                    self.skip_1x1[l], self.res_1x1[l])
+                output = torch.cat([output_buffer[l], output], dim=2)
+                output_buffer_next.append(output[:, :, -buffer_size[l]:])
+                skip_connections.append(skip)
+
+            # update buffer
+            output_buffer = output_buffer_next
+
+            # get predicted sample
+            output = sum(skip_connections)
+            output = self._postprocess(output)[0]
+            if mode == "sampling":
+                posterior = F.softmax(output[-1], dim=0)
+                dist = torch.distributions.Categorical(posterior)
+                sample = dist.sample().unsqueeze(0)
+            elif mode == "argmax":
+                sample = output.argmax(-1)
+            else:
+                logging.error("mode should be sampling or argmax")
+                sys.exit(1)
+            samples = torch.cat([samples, sample], dim=0)
+
+            # show progress
+            if interval is not None and (i + 1) % interval == 0:
+                logging.info("%d/%d estimated time = %.3f sec (%.3f sec / sample)" % (
+                    i + 1, n_samples,
+                    (n_samples - i - 1) * ((time.time() - start) / interval),
+                    (time.time() - start) / interval))
```

Comment:
How about assigning the value to a variable, e.g., `elapsed_time_par_sample = (time.time() - start) / interval`.